pytest_xmlcomp.py

```diff
@@ -7,57 +7,74 @@
 import json
 from lxml import etree
 
+def pytest_collect_file(parent, path):
+    jsonfile = path.new(ext = ".json")
+    if path.ext == ".xml" and jsonfile.exists():
+        print("XML file %s and JSON file %s found" % (path, jsonfile))
+        return XMLJSONFile(path, parent)
 
-def pytest_addoption(parser):
-    group = parser.getgroup('xmlcomp')
-    group.addoption(
-        '--datadir',
-        action='store',
-        dest='datadir',
-        default='tests/data',
-        help='Give the directory containing the XML files.')
-
-    parser.addini('HELLO', 'Dummy pytest.ini setting')
-
-
-@pytest.fixture
-def bar(request):
-    return request.config.option.datadir
-
-
-@pytest.fixture
-def xmljsonfiles(request):
-    result = []
-    # create Pathlib object for the given data directory
-    p = local(request.config.option.datadir)
-    # iterate over the file list
-
-    def xml_filter(x):
-        if x.ext == ".xml":
-            return x
-    for f in p.listdir(fil=xml_filter):
-        json = f.new(ext=".json")
-        if not json.exists():
-            raise ValueError("JSON file not found!")
-        result.append((f, json))
-    return result
-
-
-@pytest.fixture
-def compare_xml_with_json(request):
-    # create Pathlib object for the given data directory
-    p = local(request.config.option.datadir)
-    # iterate over Pathlib object
-
-    def xml_filter(x):
-        if x.ext == ".xml":
-            return x
-    for f in p.listdir(fil=xml_filter):
-        root = etree.parse(str(f))
-        j = f.new(ext=".json")
-        jsondata = json.load(open(str(j)))
-    for xpath, expresult in jsondata:
-        res = root.xpath(xpath)
-        if not res:
-            raise ValueError("XPath is not in XML file!")
-    return True
+def stringifylist(res):
+        result = []
+        for obj in res:
+            if isinstance(obj, (etree._Comment, etree._ProcessingInstruction)):
+                result.append(str(obj))
+            elif isinstance(obj, etree._ElementTree):
```

Comment:
Check if the `.xpath()` method returns an `_ElementTree` object. I think, it does not, so you could remove these two lines.